libkbfs/journal_server.go

```diff
@@ -167,25 +221,34 @@ func (j *JournalServer) Enable(
 		}
 	}()
 
-	j.lock.Lock()
-	defer j.lock.Unlock()
-
 	if _, ok := j.tlfJournals[tlfID]; ok {
 		j.log.CDebugf(ctx, "Journal already enabled for %s", tlfID)
 		return nil
 	}
 
-	if j.dirtyOps > 0 {
-		return fmt.Errorf("Can't enable journal for %s while there "+
-			"are outstanding dirty ops", tlfID)
-	}
-	if j.delegateDirtyBlockCache.IsAnyDirty(tlfID) {
-		return fmt.Errorf("Can't enable journal for %s while there "+
-			"are any dirty blocks outstanding", tlfID)
+	err = func() error {
+		if j.dirtyOps > 0 {
```

Comment:
I can't think of an easy way to handle switching users while there are dirty ops in flight. Any ideas?